contributors/design-proposals/kubeadm-upgrades.md

```diff
@@ -0,0 +1,350 @@
+     - Example: Upgrading from `v1.6.2` to `v1.8.4` is not supported.
+     - Note: We might relax this policy to allow two minor versions before going to GA.
+ - The target upgrade version must not be an alpha, beta version unless the user explicitely allows it with `--allow-experimental-upgrades`
+   - This is a **skippable policy**, in order to have an extra safety check but still allow it for users who may want it. Such upgrades are not supported.
+     - Example: Upgrading from `v1.7.1` to `v1.8.0-beta.3` can only be done with at least one of the `--allow-experimental-upgrades` or `--force` flags.
+ - The target upgrade version must not be a release candidate version unless the user explicitely allows it with `--allow-release-candidate-upgrades` (or `--allow-experimental-upgrades`)
+   - This is a **skippable policy**, in order to have an extra safety check but still allow users to easily test our RCs. These upgrades "should work".
+     - Example: Upgrading from `v1.7.1` to `v1.8.0-rc.1` can only be done with at least one of the `--allow-release-candidate-upgrades`, `--allow-experimental-upgrades` or `--force` flags.
+
+## Initial scope of `kubeadm upgrade`
+
+The initial scope of kubeadm upgrades in v1.8 is to upgrade the _control plane_ along with kube-proxy and kube-dns that are hosted on top of Kubernetes.
+The control plane upgrade is atomic; either all or nocontrol plane components get upgraded. If a post-upgrade step fails though (like upgrading Bootstrap Tokens or applying a new kube-dns), kubeadm upgrade just returns a non-zero code instead of reverting the successful control plane upgrade. We might change the latter behavior in the future.
+
+In the first iteration, upgrading the kubelets and/or etcd in the cluster is a manual task (see the sections below).
+We will address these limitations in future versions before graduating this functionality to GA.
+
+## Command structure
+
+ - `kubeadm upgrade`: Placeholder command for other subcommands.
+   - Relevant, generic flags:
+     - `--config`: A path to a configuration file to use as the desired state for the cluster component configuration. If omitted, configuration is read from a ConfigMap in the cluster.
+     - `--kubeconfig`: The path to a configuration file to use when talking to the cluster. Default: `/etc/kubernetes/admin.conf`.
+     - `--allow-experimental-upgrades`: Allow the user to upgrade to alpha, beta and release candidate versions.
+     - `--allow-release-candidate-upgrades`: Allow the user to upgrade to release candidate versions.
+     - `--print-config`: Print out the configuration read from the file or ConfigMap in the cluster.
+     - `--skip-preflight-checks`: Skip the preflight checks that are run before execution.
+ - `kubeadm upgrade plan`: Serves as a way to show users:
+   - If the cluster is in an upgradeable state.
+   - What versions the user can upgrade the cluster to.
+ - `kubeadm upgrade apply [version]`:
```

Comment:
per the sample output below this seems like it should be `kubeadm upgrade apply [--version VERSON]`

Reply:
I'll paste some real output here in a minute. It should be `kubeadm upgrade apply [version]` as people thought an extra `--version` arg was unnecessary